src/link/MachO.zig

```diff
@@ -985,15 +1007,28 @@ pub fn updateDecl(self: *MachO, module: *Module, decl: *Module.Decl) !void {
             .n_value = addr,
         };
         self.offset_table.items[decl.link.macho.offset_table_index] = addr;
-
-        try self.writeSymbol(decl.link.macho.local_sym_index);
         try self.writeOffsetTableEntry(decl.link.macho.offset_table_index);
     }
 
+    // Perform PIE fixups (if any)
+    const got_section = self.sections.items[self.got_section_index.?];
+    while (self.pie_fixups.popOrNull()) |fixup| {
+        const target_addr = fixup.address;
+        const this_addr = symbol.n_value + fixup.start;
+        if (self.base.options.target.cpu.arch == .x86_64) {
```

Comment:
This if-else would be better as a switch that had `else => unreachable` to catch bugs